# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jsapi.cpp
# Commit: 33dcf5c2e643
# Full Hash: 33dcf5c2e643b94e90a0a197c99112a080501a12
# Author: Alexandra Borovova <aborovova@mozilla.com>
# Date: 2025-08-20 09:57:21
# Regressor Bug: 1980211
# File Overlap Count: 2
# Description:
#   Bug 1980211 - Attach locale override to Realm instead of setting it in the process. r=dom-core,smaug,tschuster
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D260169
# ==============================================================================

diff -r 2cb7ca176216 -r 33dcf5c2e643 js/src/jsapi.cpp
--- a/js/src/jsapi.cpp	Tue Aug 19 16:57:59 2025 +0000
+++ b/js/src/jsapi.cpp	Tue Aug 19 17:01:50 2025 +0000
@@ -1803,6 +1803,10 @@
   return cx->realm()->behaviors();
 }
 
+void JS::SetRealmLocaleOverride(Realm* realm, const char* locale) {
+  realm->setLocaleOverride(locale);
+}
+
 void JS::SetRealmNonLive(Realm* realm) { realm->setNonLive(); }
 
 void JS::SetRealmReduceTimerPrecisionCallerType(Realm* realm,